### ISSUE TITLE
Scrolling with `<C-b>` does not always make virtual lines above first buffer visible

### ISSUE THREAD

**Issue by echasnovski** [neovim/neovim#27967]
### Problem

Virtual lines above single line in a buffer are not always made visible after `<C-b>`.

### Steps to reproduce

1. Create an 'init.lua':
    ```lua
    local ns_id = vim.api.nvim_create_namespace('virt-lines-in-single-line')
    vim.api.nvim_buf_set_lines(0, 0, -1, false, { 'single line' })
    vim.api.nvim_buf_set_extmark(0, ns_id, 0, 0, { virt_lines = { { { 'Above', 'String' } } }, virt_lines_above = true })
    ```
1. `nvim --clean -u path/to/init.lua`. The virtual line above is not visible, which seems to be expected (judging by #16166 and [this comment](https://github.com/neovim/neovim/pull/15351#issuecomment-902107285)).
2. Press `<C-b>`. Nothing happens.
3. Press `<C-y>`. Virtual line appears.

The line can also be made visible with `<C-b>` if there are at least two buffer lines.

### Expected behavior

Pressing `<C-b>` should consistently result into making virtual lines above first line visible. 

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2657+g9765efb40

### Vim (not Nvim) behaves the same?

No (doesn't have this functionality)

### Operating system/version

EndeavourOS Linux x86_64 (6.8.1-arch1-1)

### Terminal name/version

st-0.9

### $TERM environment variable

st-256color

### Installation

appimage

**Comment by luukvbaal** [comment]
Same issue is present in Vim. I think ideally `onepage()` should just call `scrollup()/scrolldown()` so that it adjusts for filler lines and 'smoothscroll' properly.

**Comment by luukvbaal** [comment]
I proposed a Vim patch. Also worth noting is that Vim shows a smoothscroll marker `<<<` when there are hidden virtual text properties above the topline. Not sure if that is intended, or just an unintended consequence virtual text properties affecting skipcol in Vim (as opposed to topfill for Nvim). It seems useful when there are hidden virtual lines above the first line of the buffer, do we want to show it then? I'm not sure its wanted for lines that are not the first line of the buffer.

**Comment by echasnovski** [comment]
> It seems useful when there are hidden virtual lines above the first line of the buffer, do we want to show it then? I'm not sure its wanted for lines that are not the first line of the buffer.

As for me - yes. As a user, I'd want to have an indicator that there are lines above the first line and *only* above the first line.

**Comment by luukvbaal** [comment]
#28044 fixes this for `<C-b>`. `<C-u>` could use the same treatment.

**Comment by zeertzjq** [comment]
Fixed by #28044
